packages/squiggle-lang/src/public/SqProject/ProjectItem.ts

```diff
@@ -8,32 +8,37 @@ import { ImmutableMap } from "../../utility/immutableMap.js";
 import * as Result from "../../utility/result.js";
 import { Ok, result } from "../../utility/result.js";
 import { Value } from "../../value/index.js";
-import {
-  SqCompileError,
-  SqError,
-  SqOtherError,
-  SqRuntimeError,
-} from "../SqError.js";
-import { Resolver } from "./Resolver.js";
+import { SqCompileError, SqError, SqRuntimeError } from "../SqError.js";
+import { SqLinker } from "../SqLinker.js";
 
-// source -> ast -> imports -> bindings & result
+// source -> ast -> imports -> result/bindings/exports
 
 export type RunOutput = {
   result: Value;
   bindings: Bindings;
+  exports: Bindings;
 };
 
-export type ImportBinding = {
-  sourceId: string;
-  variable: string;
-};
+export type Import =
+  | {
+      // For now, only `continues` can be flattened, but this might change in the future.
+      // Also, because of this, flat imports import _all_ bindings, not just exports. In the future, we might prefer to change that,
+      // and allow flat imports of exports only (for `import "foo" as *` syntax).
+      type: "flat";
```

Comment:
do we want to continue supporting flat imports? It seems that in SquiggleHub, only named imports are allowed.

Reply:
(That said, I could understand that we might be reliant on flat imports for relative values or similar - maybe we could set aside a story to clean that up later.)

Reply:
> do we want to continue supporting flat imports?

If we want to bring back Observable eventually, we'll need them. A bit hard/awkward to maintain, but still easier than to remove and reimplement later.